lib/src/code_field/span_builder.dart

```diff
@@ -38,10 +46,55 @@ class SpanBuilder {
     required Node node,
     required CodeThemeData? theme,
   }) {
+    _updateLineIndex(node);
```

Comment:
This should only be called on a node with `value` and not recursively.
And likely after this node is built.

Reply:
Value already checks on null at `node.getNewlineCount` 
Why not recursively? Children also can contain value